tracer/src/Datadog.Trace/AppSec/Waf/Initialization/LibraryLocationHelper.cs

```diff
@@ -30,16 +30,23 @@ internal static List<string> GetDatadogNativeFolders(FrameworkDescription framew
 
             List<string> paths = new();
 
-            AddNativeLoaderEnginePath(paths);
-
-            foreach (var runtimeId in runtimeIds)
+            try
             {
-                AddHomeFolders(paths, runtimeId);
-            }
+                AddNativeLoaderEnginePath(paths);
+
+                foreach (var runtimeId in runtimeIds)
+                {
+                    AddHomeFolders(paths, runtimeId);
+                }
 
-            foreach (var runtimeId in runtimeIds)
+                foreach (var runtimeId in runtimeIds)
+                {
+                    AddProfilerFolders(paths, frameworkDescription, runtimeId);
+                }
+            }
+            catch (Exception e) when (e is ArgumentException or PathTooLongException)
             {
-                AddProfilerFolders(paths, frameworkDescription, runtimeId);
+                Log.Warning(e, "When adding paths from native loader engine, home folders, and profiler folders, some path contained invalid characters or were too long");
```

Comment:
This seems very surprising 🤔 I think we should try to understand _why_ this happens. After all, the profiler runs from exactly the same folder - is this a case of manual-only instrumentation or something? Is there anything we can determine from the telemetry configuration as to what's triggering this?

Reply:
indeed that's quite strange, but as we do at some point `Path.GetDirectoryName(value)` maybe, profiler still manages to load from a path with invalid chars, but calling this method would throw 🤔